tests/common.py

```diff
@@ -0,0 +1,2 @@
+def graphs_equal(G, H):
```

Comment:
IMO it would be less confusing to a reader to do
```
self.assertIs(type(G), type(H))
self.assertTrue(nx.graphs_equal(G, H))
```
in each of the places we use this method. For users coming from `NetworkX` they'll likely to not immediately notice that this is a different method. IMO the slightly higher verbosity is worth it.

If we do keep this, needs an apache header.

Reply:
I wrote this abomination because we're still testing networkx2.4, which doesn't have a graphs_equal method :upside_down_face:

Reply:
In that case I'd rather raise the floor. High time we require 3.0

Reply:
But you give me a better idea here: to subclass UnitTest and make each equality check an assertEqual.  Then we'll have a strictly better `self.assertGraphsEqual` method to use throughout the library.

Reply:
IMO more explicit would be a mixin rather than a intermediate class.
```
class GraphMixin:
    def assertGraphsEqual(...): ...
```

Reply:
I guess an intermediate class would allow you to just use `self.assertIs(...)` etc without messing about with abstract methods. So maybe my first idea wasn't my best.

Reply:
I proposed an intermediate class and started writing it and realized that a mixin totally works -- no need for abstract methods :)